base/est/src/main/java/org/dogtagpki/est/ESTEngine.java

```diff
@@ -0,0 +1,105 @@
+package org.dogtagpki.est;
+
+import java.io.File;
+import java.io.FileReader;
+import java.util.Properties;
+
+import javax.servlet.ServletContextEvent;
+import javax.servlet.ServletContextListener;
+import javax.servlet.annotation.WebListener;
+
+/**
+ * Engine that manages the EST backend(s) according to configuration.
+ *
+ * @author Fraser Tweedale
+ */
+@WebListener
+public class ESTEngine implements ServletContextListener {
+
+    private static org.slf4j.Logger logger = org.slf4j.LoggerFactory.getLogger(ESTEngine.class);
+
+    private static ESTEngine INSTANCE;
+
+    private ESTBackend backend;
+
+    public static ESTEngine getInstance() {
+        return INSTANCE;
+    }
+
+    public ESTEngine() {
+        INSTANCE = this;
+    }
+
+    public ESTBackend getBackend() {
+        return backend;
+    }
+
+    public void start(String contextPath) throws Throwable {
+        logger.info("Starting EST engine");
+
+        String contextPathDirName = "".equals(contextPath) ? "ROOT" : contextPath.substring(1);
+        String catalinaBase = System.getProperty("catalina.base");
+        String serverConfDir = catalinaBase + File.separator + "conf";
+        String estConfDir = serverConfDir + File.separator + contextPathDirName;
+
+        logger.info("EST configuration directory: " + estConfDir);
+
+        initBackend(estConfDir + File.separator + "backend.conf");
+
+        logger.info("EST engine started");
+    }
+
+    public void stop() throws Throwable {
+        logger.info("Stopping EST engine");
+
+        if (backend != null) {
+            backend.stop();
+        }
+
+        logger.info("EST engine stopped");
+    }
```

Comment:
These methods could be merged with the context initialization and destruction, is there a specific reason to separate them? They need to be public?